src/components/config.cpp

```diff
@@ -208,6 +211,59 @@ unsigned long long config::convert(string&& value) const {
   return v < ULLONG_MAX ? v : 0ULL;
 }
 
+template <>
+spacing_val config::convert(string&& value) const {
+  char* new_end;
+  auto size_value = std::strtof(value.c_str(), &new_end);
+
+  if (size_value < 0) {
+    throw application_error(sstream() << "Value: " << value << " must be positive ");
+  }
+
+  spacing_val size{spacing_type::SPACE, size_value};
+
+  string unit = string_util::trim(new_end);
+  if (!unit.empty()) {
+    if (unit == "px") {
+      size.type = spacing_type::PIXEL;
+      size.value = std::trunc(size.value);
+    } else if (unit == "pt") {
+      size.type = spacing_type::POINT;
+    } else {
+      size.value = std::trunc(size.value);
+    }
+  }
+
+  return size;
```

Comment:
If we have spaces, the number of spaces is not truncated and may be a floating point number.